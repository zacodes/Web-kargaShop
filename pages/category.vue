<template>
  <div class="container my-5" id="Home">
    <div class="row">
      <div class="col-3">
        <div class="row">
          <div class="col-12">
            <h3 class="text-ml">Ürün Grupları</h3>
            <hr class="head-border float-left">
            <div class="clearfix"></div>
            Seçimleriniz <br><br>
            Filtre Seçenekleri
            <div class="col-12 border-bottom my-4"></div>
            <div id="uni" class="font-weight-bold">UNISEX T-SHIRT</div>
            <a href="" style="color:#efc002" class="d-block mt-3">
              <i class="fa fa-angle-left"></i> Giyim
            </a>
            <div class="col-12 border-bottom my-4"></div>
            <div id="mark" class="font-weight-bold text-left">
              Tüm Markalar
            </div>

            <div class="form-check">
              <input class="form-check-input" type="checkbox" value="" id="defaultCheck1">
              <label id="kar" class="form-check-label" for="defaultCheck1">
                Karga
              </label>
            </div>

            <div id="filt" class="font-weight-bold text-left mt-5">
              Filtre Seçenekleri
            </div>

            <div class="form-check">
              <input class="form-check-input" type="checkbox" value="" id="defaultCheck2">
              <label id="kamp" class="form-check-label" for="defaultCheck2">
                Kampanyalı Ürünler
              </label>
            </div>
            <div class="form-check">
              <input class="form-check-input" type="checkbox" value="" id="defaultCheck3">
              <label id="spons" class="form-check-label" for="defaultCheck3">
                Sponsor Ürünler
              </label>
            </div>
            <div class="form-check">
              <input class="form-check-input" type="checkbox" value="" id="defaultCheck4">
              <label id="yeni" class="form-check-label" for="defaultCheck4">
                Yeni Ürünler
              </label>
            </div>


            <div id="enp" class="font-weight-bold text-left mt-5">
              En Popüler Olanlar
            </div>

            <a href="/product" class="d-block">
              <i class="fa fa-camera"></i>
              KANAGAWA -UNISEX TSHIRT
            </a>
          </div>
        </div>
      </div>
      <div class="col-9">
        <div class="row">
          <div class="col-3 product-card-container">
            <b-link to="/product">
              <b-card
                :title="product.name"
                :img-src="product.images[0]"
                :img-alt="product.name"
                img-top
                tag="article"
                style="max-width: 20rem;"
                class="mb-2"
              >
                <b-card-text>
                  {{ product.name }}
                </b-card-text>
                <p class="card-price font-weight-bold">
                  {{ product.price }} TL
                </p>
              </b-card>
            </b-link>
          </div>
        </div>
      </div>
    </div>
  </div>
</template>

<script>
export default {
  data () {
    return {
      product: {
        name: 'KANAGAWA -UNISEX TSHIRT',
        category: 'Unisex T-Shirt',
        price: 34.9,
        code: 'K1867',
        sizes: [
          'S', 'M', 'L', 'XL'
        ],
        images: [
          '_nuxt/assets/image/product1.jpg'
        ]
      }
    }
  }
  
}
</script>

