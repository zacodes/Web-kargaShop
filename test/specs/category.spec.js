import { mount, shallowMount } from '@vue/test-utils'
import category from "../../pages/category.vue"


describe('category.vue', () => {

    it('category.vue Bileşenimizin class özelliğini kontrol edilmesi', () => {
        const wrapper = mount(category)
        expect(wrapper.classes()).toContain('container', 'my-5');
    });

    it("test1", () => {
        const wrapper = mount(category)
        expect(wrapper.text()).toContain("Seçimleriniz");
    });

    it("test2", () => {
        const wrapper = mount(category)
        expect(wrapper.text()).toContain("Ürün Grupları");
    });

    it("test3", () => {
        const wrapper = mount(category)
        expect(wrapper.text()).toContain("Filtre Seçenekleri");
    });

    it("test4", () => {
        const wrapper = mount(category)
        expect(wrapper.text()).toContain("UNISEX T-SHIRT");
    });

    it("test5", () => {
        const wrapper = mount(category)
        expect(wrapper.text()).toContain("Giyim");
    });

    it("test6", () => {
        const wrapper = mount(category)
        expect(wrapper.text()).toContain("Tüm Markalar");
    });

    it("test7", () => {
        const wrapper = mount(category)
        expect(wrapper.text()).toContain("Karga");
    });

    it("test8", () => {
        const wrapper = mount(category)
        expect(wrapper.text()).toContain("Filtre Seçenekleri");
    });

    it("test9", () => {
        const wrapper = mount(category)
        expect(wrapper.text()).toContain("Kampanyalı Ürünler");
    });

    it("test9", () => {
        const wrapper = mount(category)
        expect(wrapper.text()).toContain("Kampanyalı Ürünler");
    });

    it("test10", () => {
        const wrapper = mount(category)
        expect(wrapper.text()).toContain("Sponsor Ürünler");
    });

    it("test11", () => {
        const wrapper = mount(category)
        expect(wrapper.text()).toContain("Yeni Ürünler");
    });

    it("test12", () => {
        const wrapper = mount(category)
        expect(wrapper.text()).toContain("En Popüler Olanlar");
    });

    it("test13", () => {
        const wrapper = mount(category)
        expect(wrapper.text()).toContain("KANAGAWA -UNISEX TSHIRT");
    });

    it('Snapshot özelliği', () => {
        const wrapper = mount(category)
        expect(wrapper.html()).toMatchSnapshot()
    });

    it('has data', () => {
        const wrapper = mount(category)
    expect(typeof category.data).toBe('function')
    });
})
