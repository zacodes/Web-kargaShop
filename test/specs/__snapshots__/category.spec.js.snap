// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`category.vue Snapshot özelliği 1`] = `
<div id="Home" class="container my-5">
  <div class="row">
    <div class="col-3">
      <div class="row">
        <div class="col-12">
          <h3 class="text-ml">Ürün Grupları</h3>
          <hr class="head-border float-left">
          <div class="clearfix"></div>
          Seçimleriniz <br><br>
          Filtre Seçenekleri
          <div class="col-12 border-bottom my-4"></div>
          <div id="uni" class="font-weight-bold">UNISEX T-SHIRT</div> <a href="" class="d-block mt-3" style="color: rgb(239, 192, 2);"><i class="fa fa-angle-left"></i> Giyim
          </a>
          <div class="col-12 border-bottom my-4"></div>
          <div id="mark" class="font-weight-bold text-left">
            Tüm Markalar
          </div>
          <div class="form-check"><input type="checkbox" value="" id="defaultCheck1" class="form-check-input"> <label id="kar" for="defaultCheck1" class="form-check-label">
              Karga
            </label></div>
          <div id="filt" class="font-weight-bold text-left mt-5">
            Filtre Seçenekleri
          </div>
          <div class="form-check"><input type="checkbox" value="" id="defaultCheck2" class="form-check-input"> <label id="kamp" for="defaultCheck2" class="form-check-label">
              Kampanyalı Ürünler
            </label></div>
          <div class="form-check"><input type="checkbox" value="" id="defaultCheck3" class="form-check-input"> <label id="spons" for="defaultCheck3" class="form-check-label">
              Sponsor Ürünler
            </label></div>
          <div class="form-check"><input type="checkbox" value="" id="defaultCheck4" class="form-check-input"> <label id="yeni" for="defaultCheck4" class="form-check-label">
              Yeni Ürünler
            </label></div>
          <div id="enp" class="font-weight-bold text-left mt-5">
            En Popüler Olanlar
          </div> <a href="/product" class="d-block"><i class="fa fa-camera"></i>
            KANAGAWA -UNISEX TSHIRT
          </a>
        </div>
      </div>
    </div>
    <div class="col-9">
      <div class="row">
        <div class="col-3 product-card-container">
          <b-link to="/product">
            <b-card title="KANAGAWA -UNISEX TSHIRT" img-src="_nuxt/assets/image/product1.jpg" img-alt="KANAGAWA -UNISEX TSHIRT" img-top="" tag="article" class="mb-2" style="max-width: 20rem;">
              <b-card-text>
                KANAGAWA -UNISEX TSHIRT
              </b-card-text>
              <p class="card-price font-weight-bold">
                34.9 TL
              </p>
            </b-card>
          </b-link>
        </div>
      </div>
    </div>
  </div>
</div>
`;
